### PR TITLE
Doco update for Groups

### PR DESCRIPTION
**Pull request by ryan-royals**
Found through trial and error that both `RoleManagement.ReadWrite.Directory` and `Directory.ReadWrite.All` are required for `assignable_to_role` property.

### CODE REVIEW

**Comment by ryan-royals** [comment]
@manicminer @katbyte Sorry for the random tag.
Not sure the process to getting this merged, but it's a minor doco change that has been hanging for ages.

**Comment by manicminer** [comment]
@ryan-royals Thanks for the PR and apologies for the delay - I would like to manually test this before merging as permissions for groups are tricky and don't align well with the official API documentation.

**Comment by manicminer** [comment]
@ryan-royals Sorry for the delay in getting back to this PR. I've tested with a new service principal and it looks like `Directory.ReadWrite.All` does nothing in this context, and `RoleManagement.ReadWrite.Directory` is the role that's needed.

File: docs/resources/group.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ When authenticated with a service principal, this resource requires one of the f
 
 Alternatively, if the authenticated service principal is also an owner of the group being managed, this resource can use the application role: `Group.Create`.
 
-If using the `assignable_to_role` property, this resource additionally requires one of the following application roles: `RoleManagement.ReadWrite.Directory` or `Directory.ReadWrite.All`
+If using the `assignable_to_role` property, this resource additionally requires both of the following application roles: `RoleManagement.ReadWrite.Directory` and `Directory.ReadWrite.All`
```
Comment (manicminer) [inline docs/resources/group.md:17]:
```suggestion
If using the `assignable_to_role` property, this resource additionally requires the `RoleManagement.ReadWrite.Directory` application role.
```

**Review by manicminer — APPROVED** [review]

**Pull request merged**